_resources/port1.0/group/python-1.0.tcl

```diff
@@ -270,22 +273,64 @@ proc python_set_default_version {option action args} {
 
 options python.branch python.prefix python.bin python.lib python.libdir \
         python.include python.pkgd
-# for pythonXY, python.branch is X.Y
-default python.branch   {[string range ${python.version} 0 end-1].[string index ${python.version} end]}
-default python.prefix   {${frameworks_dir}/Python.framework/Versions/${python.branch}}
-default python.bin      {${python.prefix}/bin/python${python.branch}}
-default python.lib      {${python.prefix}/Python}
-default python.pkgd     {${python.prefix}/lib/python${python.branch}/site-packages}
-default python.libdir   {${python.prefix}/lib/python${python.branch}}
+# for pythonXY or pypyXY, python.branch is X.Y
+default python.branch   {[string index ${python.version} end-1].[string index ${python.version} end]}
+default python.prefix   {[python_get_defaults prefix]}
+default python.bin      {[python_get_defaults bin]}
+default python.lib      {[python_get_defaults lib]}
+default python.pkgd     {[python_get_defaults pkgd]}
+default python.libdir   {[python_get_defaults libdir]}
 default python.include  {[python_get_defaults include]}
 default build.cmd       {${python.bin} setup.py --no-user-cfg}
 default destroot.cmd    {${python.bin} setup.py --no-user-cfg}
 default destroot.destdir {--prefix=${python.prefix} --root=${destroot}}
 
 proc python_get_defaults {var} {
-    global python.version python.branch python.prefix
+    global frameworks_dir prefix python.version python.branch python.prefix
     switch -- $var {
+        prefix {
+            if {[string match py* ${python.version}]} {
+                return ${prefix}/lib/py${python.version}
+            } else {
+                return ${frameworks_dir}/Python.framework/Versions/${python.branch}
+            }
+        }
+        bin {
+            if {[string match py3? ${python.version}]} {
+                return ${python.prefix}/bin/pypy3
+            } elseif {[string match py* ${python.version}]} {
+                return ${python.prefix}/bin/pypy
+            } else {
+                return ${python.prefix}/bin/python${python.branch}
+            }
+        }
+        lib {
+            if {[string match py3? ${python.version}]} {
+                return ${python.prefix}/bin/libpypy3-c.dylib
+            } elseif {[string match py* ${python.version}]} {
+                return ${python.prefix}/bin/libpypy-c.dylib
+            } else {
+                return ${python.prefix}/Python
+            }
+        }
+        pkgd {
+            if {[string match py* ${python.version}]} {
+                return ${python.prefix}/site-packages
+            } else {
+                return ${python.prefix}/lib/python${python.branch}/site-packages
+            }
+        }
+        libdir {
+            if {[string match py3? ${python.version}]} {
+                return ${python.prefix}/bin/libpypy3-c.dylib
```

Comment:
This seems like a nonsensical value for libdir.

Reply:
Yeah, fixed.